### ISSUE TITLE
Heroic Quad Cannon can never be disabled by ECM Tanks.

### ISSUE THREAD

**Issue by Jundiyy** [TheSuperHackers/GeneralsGamePatch#396]
A Vet 3 Quad Cannon can never be shut down by ECMs, no matter how many you have.

A simple fix that I learnt from Predator quite some time ago is to increase the SubdualDamageCap

Current stats are as follows.

    SubdualDamageCap = 440
    SubdualDamageHealRate = 500
    SubdualDamageHealAmount = 50

By changing it to this SubdualDamageCap = 450 it fixed the problem and I didn't see any gameplay changes, of course I'd prefer others to test this as well just in case.

Video to help understand in case someone doesn't.

https://youtu.be/Vw_2uhyoVwk

**Comment by xezon** [comment]
Yes this is already tracked in commy2 list
https://github.com/commy2/zerohour/issues/10

**Comment by Jundiyy** [comment]
Oh, I did search for it but it didn't come up, now I see why, it's not under your git.

Okay so there is a small difference,
I know he said this `This bug can be fixed by setting the SubdualDamageCap entry to 600 as intended.` but if there is no other benefit other than keeping it original, the fix looked like it was done with setting it to 450 (still waiting on others to test).

The difference between the 2 shows when they wake back up.

https://youtu.be/MEV0eDAEsS8

**Comment by Jundiyy** [comment]
Oh, I did search for it but it didn't come up, now I see why, it's not under your git.

Okay so there is a small difference,
I know he said this `This bug can be fixed by setting the SubdualDamageCap entry to 600 as intended.` but if there is no other benefit other than keeping it original I'd prefer the 450 fix (still waiting on others to confirm it works) as there is a small difference.

The difference between the 2 shows when they wake back up.

https://youtu.be/Vw_2uhyoVwk - SubdualDamageCap 450
https://youtu.be/MEV0eDAEsS8 - SubdualDamageCap 600

**Comment by xezon** [comment]
External tasks are tracked in our various task documents as opposed to the issues section:

https://github.com/xezon/GeneralsGamePatch/tree/main/Patch104pZH/Design/Tasks

**Comment by commy2** [comment]
I did go with 600, because that is health x2, like every other unit has. The 440 currently used is the result of the old Quad HP of 220, which was buffed to 300 in ZH 1.04. (The old Quad HP can still be found in the GC single player only Quads). I would be careful with edging for the lowest possible value, because other health buffs could still break it then.

Edit: Hold The Line and playing vs Easy AI or against Hard AI come to mind.

**Comment by Jundiyy** [comment]
Hmmm okay, well I suppose more testing to be sure? If we can get a similar match or original wake up with lower sub health, I think that would be better than having the double sub health and having different wake up times.

Sound good? Or meh 😂

**Comment by commy2** [comment]
Well, if it is decided that the Quad, for whatever reason, should be especially hardened or susceptible to ECM jamming, then the `SUBDUAL_VEHICLE` armor and the recovery rate (SubdualDamageHealAmount) could be tweaked. I just don't see the point, why the Quad Cannon should be treated any differently than other units.

The bug currently is that the total health exceeds the subdual health at vet 3 (and I think vet 2?). Normally the disabled state is reached when the subdual health falls below the health (and why it must be larger than the health), but health bonuses do not apply to subdual health.

**Comment by Jundiyy** [comment]
Vet 2 does lock down, it's only vet 3.
Yes I get that and I always thought you needed more health for it but today I tested it with less health and it worked with better results hence why I am asking. 
I know according to coding it wouldn't match with the rest of the units but in-game it will, as you can see from the second video they wake up at different times. The vet 3 wakes up a lot after the vet 2.
Check both videos to compare.

**Comment by commy2** [comment]
It seems the mechanics of ECM jamming are poorly understood and more testing would help. Not just because of the patch, but in general for understanding how the game operates.
